.github/workflows/ci.yml

```diff
@@ -57,22 +57,22 @@ jobs:
                 cargo build --release -p test-no-std --features node-api,
                 cargo build --release -p test-no-std --features primitives,
 
-                 # Check build of stand-alone features.
-                cargo check --no-default-features,
-                cargo check --no-default-features --features sync-api,
-                cargo check --no-default-features --features jsonrpsee-client,
-                cargo check --no-default-features --features tungstenite-client,
-                cargo check --no-default-features --features ws-client,
-                cargo check --no-default-features --features staking-xt,
-                cargo check --no-default-features --features contracts-xt,
-                cargo check --no-default-features --features std,
+                # Check build of stand-alone features.
+                cargo check --no-default-features -p substrate-api-client,
+                cargo check --no-default-features -p substrate-api-client --features sync-api,
+                cargo check --no-default-features -p substrate-api-client --features jsonrpsee-client,
+                cargo check --no-default-features -p substrate-api-client --features tungstenite-client,
+                cargo check --no-default-features -p substrate-api-client --features ws-client,
+                cargo check --no-default-features -p substrate-api-client --features staking-xt,
+                cargo check --no-default-features -p substrate-api-client --features contracts-xt,
+                cargo check --no-default-features -p substrate-api-client --features std,
```

Comment:
When I moved the `substreate-api-client` to its own pure package crate the default behaviour changed to executed these checks for the whole workspace. I put this change now to preserve the current behaviour as some CI checks actually fail. I decided that it is better to fix this in a follow up issue as there are also some clippy warnings.

Reply:
Yeah, I also stumbled upon this a while ago. I did not investigate the root cause exactly then, but we should fix the clippy issues 👍